homeassistant/components/microbees/config_flow.py

```diff
@@ -0,0 +1,83 @@
+"""Config flow for microBees integration."""
+from collections.abc import Mapping
+import logging
+from typing import Any
+
+import microBeesPy
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_ACCESS_TOKEN, CONF_TOKEN
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import aiohttp_client, config_entry_oauth2_flow
+
+from .const import DOMAIN
+
+VERSION = 1
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class OAuth2FlowHandler(
+    config_entry_oauth2_flow.AbstractOAuth2FlowHandler, domain=DOMAIN
+):
+    """Handle a config flow for microBees."""
+
+    DOMAIN = DOMAIN
+
+    @property
+    def logger(self) -> logging.Logger:
+        """Return logger."""
+        return logging.getLogger(__name__)
+
+    reauth_entry: config_entries.ConfigEntry | None = None
+
+    @property
+    def extra_authorize_data(self) -> dict[str, Any]:
+        """Extra data that needs to be appended to the authorize url."""
+        scopes = ["read", "write"]
+        return {"scope": " ".join(scopes)}
+
+    async def async_oauth_create_entry(self, data: dict) -> FlowResult:
+        """Create an oauth config entry or update existing entry for reauth."""
+
+        microbees = microBeesPy.microbees.MicroBees(
+            session=aiohttp_client.async_get_clientsession(self.hass),
+            token=data[CONF_TOKEN][CONF_ACCESS_TOKEN],
+        )
+
+        try:
+            current_user = await microbees.getMyProfile()
+        except microBeesPy.microbees.MicroBeesException:
+            return self.async_abort(reason="token_invalid")
```

Comment:
Why is it always token invalid? What if the service is down?

Reply:
MicroBees Exception is reached only if the response status is different from what we expected. in case of service down or internet missing we have exception catch below.

Reply:
What response statuses are expected and which arent?

Reply:
Are managed in the library. We manage in library status code 200 only, if others we raise the exception : https://github.com/microBeesTech/pythonSDK/blob/d7e4328111afc8cb1d320c184c7d6a0a1c0bdea3/microBeesPy/microbees.py#L135

Reply:
So why do we abort with "invalid_token" if there can be a lot of other reasons why the setup doesnt work?